Sources/SwiftGodot/Core/InspectableProperty.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  File.swift
+//  
+//
+//  Created by Marquis Kurt on 10/15/23.
+//
+
+import Foundation
+
+/// A structure that houses a property that can be added to a Godot inspector.
+public struct InspectableProperty<T> {
+    /// A typealias for the the method type used to register a property to Godot for inspectors.
+    public typealias RegisteredPropertyFunction = (T) -> ([Variant]) -> Variant?
+
+    /// The host object the property derives from.
+    public var hostObject: T.Type
```

Comment:
You might not need to store this on an instance, since it'll already be in the type (e.g. it'll be `InspectableProperty<Int>.T` )

Reply:
I believe I've tried this before with the stuff I've done with it, and it didn't quite work as intended. Hence why I have host object as a specific property in here.

Reply:
Can you say more about what went wrong? I'm using it like this elsewhere so it'd be good to know if something can go wrong here!

Reply:
I believe it was when I was trying to make the #autoProperty macro (ex.: `#autoProperty(Foo.self, "nerds"`). #autoProperty needs the variant's host type, and I couldn't fetch it from the parent class (since the parent declaration would be nil. Thus, I needed a location to store the host type into.

Reply:
I don't imagine this will be an issue for upstream as much, since #autoProperty isn't a thing here, though.